### PR TITLE
feat: Use justfile vs. makefile

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

- Upgrades Go dependencies
- Replaces makefile with justfile
- Cleans up and updates GitHub actions with new versions of actions
- Cleans up failing linter checks

## Steps to QA

- Run checks / scripts in justfile:
  - `just build-all`
  - `just lint`
  - `just format`
  - `just run`

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -36,47 +46,74 @@ jobs:
...
-            -ldflags="-s -w" \
-            -ldflags="-X 'github.com/open-sauced/pizza-cli/pkg/utils.writeOnlyPublicPosthogKey=${{ vars.POSTHOG_WRITE_PUBLIC_KEY }}'" \
-            -ldflags="-X 'github.com/open-sauced/pizza-cli/pkg/utils.Version=${{ needs.release.outputs.release-tag }}'" \
-            -ldflags="-X 'github.com/open-sauced/pizza-cli/pkg/utils.Sha=$(git rev-parse HEAD)'" \
-            -o build/pizza-${{ matrix.goos }}-${{ matrix.goarch }}
-        gh release upload ${{ needs.release.outputs.release-tag }} build/pizza-${{ matrix.goos }}-${{ matrix.goarch }}
+        export RELEASE_TAG_VERSION=${{ needs.release.outputs.release-tag }}
+        just build-${{ matrix.goos }}-${{ matrix.goarch }}
 
+        gh release upload ${{ needs.release.outputs.release-tag }} build/saucectl-${{ matrix.goos }}-${{ matrix.goarch }}
```
Comment (nickytonline) [inline .github/workflows/release.yaml:93]: TIL about `gh release`. 🔥

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM --platform=$BUILDPLATFORM golang:1.22.5 AS builder
```
Comment (nickytonline) [inline Dockerfile:1]: What were we doing prior to this Docker file?

**Review by nickytonline — APPROVED** [review]
LGTM, just a couple questions.

File: Dockerfile (hunk 2 above)
Comment (jpmcb) [reply Dockerfile:1]: Nothing! There was no image for this CLI. This will be net new (but was proved out in the `saucectl` proof of concept)

**Review by BekahHW — APPROVED** [review]

**Pull request merged**